src/Configuration.php

```diff
@@ -353,13 +360,29 @@ public function setAppType($type)
     }
 
     /**
+     * Set the fallback application type.
+     *
+     * This is should be used only by libraries to set an fallback app type.
+     *
+     * @param string|null $type the fallback type
+     *
+     * @return $this
+     */
+    public function setFallbackType($type)
+    {
+        $this->fallbackType = $type;
+
+        return $this;
+    }
+
+    /**
      * Get the application data.
      *
      * @return array
      */
     public function getAppData()
     {
-        return array_merge(['type' => php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
+        return array_merge(['type' => $this->fallbackType ?: php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
```

Comment:
If you call setFallbackType with null it could fall back to the default, though I think the user has said null so I would say the test is wrong.